src/main/kotlin/de/bund/digitalservice/useid/config/SecurityHeadersFilter.kt

```diff
@@ -2,50 +2,43 @@ package de.bund.digitalservice.useid.config
 
 import de.bund.digitalservice.useid.widget.WIDGET_PAGE
 import org.springframework.http.HttpHeaders
-import org.springframework.http.server.reactive.ServerHttpRequest
-import org.springframework.http.server.reactive.ServerHttpResponse
-import org.springframework.web.server.ServerWebExchange
-import org.springframework.web.server.WebFilter
-import org.springframework.web.server.WebFilterChain
+import org.springframework.http.server.PathContainer
+import org.springframework.web.filter.GenericFilterBean
 import org.springframework.web.util.pattern.PathPattern
 import org.springframework.web.util.pattern.PathPatternParser
-import reactor.core.publisher.Mono
+import javax.servlet.FilterChain
+import javax.servlet.ServletRequest
+import javax.servlet.ServletResponse
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
 
 class SecurityHeadersFilter(
     private val contentSecurityPolicyProperties: ContentSecurityPolicyProperties
-) : WebFilter {
+) : GenericFilterBean() {
```

Comment:
Should we use a "OncePerRequestFilter" here as well?

Reply:
it's a good idea, then I guess, we don't need to cast the request/response

Reply:
I've changed it in [this commit](https://github.com/digitalservicebund/useid-backend-service/pull/198/commits/d48638489d9520367c5148c53ac1636b45b641d4).

Reply:
Thank you!